project1-sps/script.js

```diff
@@ -1,4 +1,144 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+// initialise variables
+let boolUserEnteredName = 0;
+let userName = "";
+let iCountUserWins = 0;
+let iCountComputerWins = 0;
+let iCountDraws = 0;
+let iCountTotalRounds = 0;
+
+var randomIntegerGenerator = function (maxNum) {
+  let nRandomNum = Math.random() * maxNum;
+  let iRandomInt = Math.floor(nRandomNum);
+  iRandomInt += 1;
+  return iRandomInt;
+};
+
+// Helper function to retrieve computer random choice from computer's random integer
+var retrieveComputerRandomChoice = function (iRandomInt) {
+  let sComputerRandomChoice = "";
+  if (iRandomInt === 1) {
+    sComputerRandomChoice = "scissors";
+  } else if (iRandomInt === 2) {
+    sComputerRandomChoice = "paper";
+  } else if (iRandomInt === 3) {
+    sComputerRandomChoice = "stone";
+  } else {
+    sComputerRandomChoice = "error";
+  }
+
+  return sComputerRandomChoice;
+};
+
+// Helper function to compute userWin (0 means loss, 0.5 means draw, 1 means won)
+var computeUserWin = function (userInput, sComputerRandomChoice, userWin) {
+  if (userInput === "scissors" && sComputerRandomChoice === "paper") {
+    userWin = 1;
+  } else if (sComputerRandomChoice === "scissors" && userInput === "paper") {
+    userWin = 0;
+  }
+  // paper beats stone
+  else if (userInput === "paper" && sComputerRandomChoice === "stone") {
+    userWin = 1;
+  } else if (sComputerRandomChoice === "paper" && userInput === "stone") {
+    userWin = 0;
+  }
+  // stone beats scissors
+  else if (userInput === "stone" && sComputerRandomChoice === "scissors") {
+    userWin = 1;
+  } else if (sComputerRandomChoice === "stone" && userInput === "scissors") {
+    userWin = 0;
+  }
+  // If both parties choose the same object, it's a draw (assign arbitrary score of 0.5)
+  else if (sComputerRandomChoice === "scissors" && userInput === "scissors") {
+    userWin = 0.5;
+  } else if (sComputerRandomChoice === "paper" && userInput === "paper") {
+    userWin = 0.5;
+  } else if (sComputerRandomChoice === "stone" && userInput === "stone") {
+    userWin = 0.5;
+  }
```

Comment:
Could we try to shorten this by using || (OR) operators? Think something like grouping all those with the same userWin result and separating the conditions with || instead. e.g. lines 52-56 can be `else if ((com === "scissors" && user === "scissors") || (com === "paper" && user === "paper") || (com === "stone" && user === "stone"))`

Reply:
Thank you for sharing this tip Kimmi.